File: Java/AdivinarNumeroAleatorio/src/AdivinarNumeroAleatorio.java
import java.util.Random;
import java.util.Scanner;

public class AdivinarNumeroAleatorio {
    public static void main(String[] args) {
        jugarAdivinaNumero();
    }

    public static void jugarAdivinaNumero() {
        Random random = new Random();
        int numeroSecreto = random.nextInt(100) + 1;

        Scanner scanner = new Scanner(System.in);

        System.out.println("Adivina el número secreto (entre 1 y 100):");

        while (true) {
            int intento = scanner.nextInt();

            if (intento == numeroSecreto) {
                System.out.println("¡Felicidades! Adivinaste el número secreto.");
                break;
            } else if (intento < numeroSecreto) {
                System.out.println("El número secreto es mayor. Intenta nuevamente:");
            } else {
                System.out.println("El número secreto es menor. Intenta nuevamente:");
            }
        }

        scanner.close();
    }
}

File: C++/AdivinarNumeroAleatorio/AdivinarNumeroAleatorio/AdivinarNumeroAleatorio/AdivinarNumeroAleatorio.cpp
#include <iostream>
#include <random>

void jugarAdivinaNumero() {
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<int> distribucion(1, 100);
    int numeroSecreto = distribucion(gen);

    std::cout << "Adivina el numero secreto (entre 1 y 100):" << std::endl;

    while (true) {
        int intento;
        std::cin >> intento;

        if (intento == numeroSecreto) {
            std::cout << "¡Felicidades! Adivinaste el numero secreto." << std::endl;
            break;
        }
        else if (intento < numeroSecreto) {
            std::cout << "El numero secreto es mayor. Intenta nuevamente:" << std::endl;
        }
        else {
            std::cout << "El numero secreto es menor. Intenta nuevamente:" << std::endl;
        }
    }
}

int main() {
    jugarAdivinaNumero();
    return 0;
}
